components/async-status-mgt/org.wso2.carbon.identity.framework.async.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/status/mgt/internal/dao/impl/AsyncStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,544 @@
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.STATUS_MESSAGE;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.SUBJECT_ID;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.SUBJECT_TYPE;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.TARGET_ORG_ID;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.UNIT_OPERATION_ID;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.constant.SQLConstants.UPDATE_ASYNC_OPERATION;
+import static org.wso2.carbon.identity.framework.async.status.mgt.internal.util.Utils.isMSSqlDB;
+
+/**
+ * DAO implementation for Asynchronous Operation Status Management.
+ */
+public class AsyncStatusMgtDAOImpl implements AsyncStatusMgtDAO {
+
+    private static final Log LOG = LogFactory.getLog(AsyncStatusMgtDAOImpl.class);
+
+    @Override
+    public String registerAsyncStatusWithoutUpdate(OperationInitDTO record) throws AsyncStatusMgtException {
+
+        String generatedOperationId;
+        String currentTimestamp = new Timestamp(new Date().getTime()).toString();
+        String operationId = UUID.randomUUID().toString();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                     CREATE_ASYNC_OPERATION, ID)) {
+
+            statement.setString(OPERATION_ID, operationId);
+            statement.setString(CORRELATION_ID, record.getCorrelationId());
+            statement.setString(OPERATION_TYPE, record.getOperationType());
+            statement.setString(SUBJECT_TYPE, record.getOperationSubjectType());
+            statement.setString(SUBJECT_ID, record.getOperationSubjectId());
+            statement.setString(INITIATED_ORG_ID, record.getResidentOrgId());
+            statement.setString(INITIATED_USER_ID, record.getInitiatorId());
+            statement.setString(STATUS, OperationStatus.ONGOING.toString());
+            statement.setString(CREATED_AT, currentTimestamp);
+            statement.setString(LAST_MODIFIED, currentTimestamp);
+            statement.setString(POLICY, record.getOperationPolicy());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                generatedOperationId = generatedKeys.getString(1);
+            } else {
+                throw new SQLException("Creating operation failed, no ID obtained.");
+            }
+        } catch (SQLException e) {
+            throw new AsyncStatusMgtServerException("Error while adding Async Status Initial information " +
+                    "in the system.", e);
```

Comment:
do not hard code error messages in these places.
Define an server type error message with error code and use it here.
Fix similar places